src/allocator/allocator.c

```
@@ -280,31 +303,34 @@ int add_chunk_async(CUdeviceptr *address, size_t size, CUstream hStream) {
+}
+
+int add_chunk_from_pool_async(CUdeviceptr *address, size_t size, CUmemoryPool pool, CUstream hStream) {
+    size_t addr=0;
+    CUresult res = CUDA_SUCCESS;
+    CUdevice dev;
+    cuCtxGetDevice(&dev);
+    if (oom_check(dev,size))
+        return CUDA_ERROR_OUT_OF_MEMORY;
+
+    allocated_list_entry *e;
+    INIT_ALLOCATED_LIST_ENTRY(e, addr, size, dev);
```

Cppcheck (2.21.0)

[error] Memory leak

(memleak)